### PR TITLE
Doc update

### PR DESCRIPTION
**Pull request by gpotter2**
- a few doc fixes / improvements of build process
- add RFC-like templates in the autodoc. They are expandable and minimized by default
99C8F617-BFE8-4182-AB95-B2317E17EA02

### CODE REVIEW

File: scapy/__init__.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
 Scapy: create, send, sniff, dissect and manipulate network packets.
 
 Usable either from an interactive console or as a Python library.
-http://www.secdev.org/projects/scapy
+https://scapy.net
```
Comment (guedou) [inline scapy/__init__.py]: Is this change needed?

Comment (p-l-) [reply scapy/__init__.py]: I think it's better to point to the website rather than to the repository, so I'd say yes.

**Comment by gpotter2** [comment]
I added comments and a basic test (it will be tested through the docs anyways). I don't see the point of splitting this function

**Comment by guedou** [comment]
The comments help a lot. Thank you!

I still believe that the function should be simplified: it does not fit on single screen, and its internal logic cannot be tested independently.

Could you add some real outputs to the unit test? For example, `rfc(Ether)`, and `rfc(IPv6)`. If we merge the code as it is, this will simplify tracking regressions.

**Comment by gpotter2** [comment]
I've added real tests.
Do you still want to reduce the function ?

**Comment by guedou** [comment]
@gpotter2 sorry I did not see your message earlier. A shorter function will be better, but I am fine with the comments and the tests. Thanks!
